exporter/exporterhelper/logs_batch.go

```diff
@@ -0,0 +1,136 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package exporterhelper // import "go.opentelemetry.io/collector/exporter/exporterhelper"
+
+import (
+	"context"
+	"errors"
+
+	"go.opentelemetry.io/collector/exporter/exporterbatcher"
+	"go.opentelemetry.io/collector/pdata/plog"
+)
+
+// mergeLogs merges two logs requests into one.
+func mergeLogs(_ context.Context, r1 Request, r2 Request) (Request, error) {
+	lr1, ok1 := r1.(*logsRequest)
+	lr2, ok2 := r2.(*logsRequest)
+	if !ok1 || !ok2 {
+		return nil, errors.New("invalid input type")
+	}
+	lr2.ld.ResourceLogs().MoveAndAppendTo(lr1.ld.ResourceLogs())
+	return lr1, nil
+}
+
+// mergeSplitLogs splits and/or merges the logs into multiple requests based on the MaxSizeConfig.
+func mergeSplitLogs(_ context.Context, cfg exporterbatcher.MaxSizeConfig, r1 Request, r2 Request) ([]Request, error) {
+	var (
+		res          []Request
+		destReq      *logsRequest
+		capacityLeft = cfg.MaxSizeItems
+	)
+	for _, req := range []Request{r1, r2} {
+		if req == nil {
+			continue
+		}
+		srcReq, ok := req.(*logsRequest)
+		if !ok {
+			return nil, errors.New("invalid input type")
+		}
+		if srcReq.ld.LogRecordCount() <= capacityLeft {
+			if destReq == nil {
+				destReq = srcReq
+			} else {
+				srcReq.ld.ResourceLogs().MoveAndAppendTo(destReq.ld.ResourceLogs())
+			}
+			capacityLeft = cfg.MaxSizeItems - destReq.ld.LogRecordCount()
+			continue
+		}
+
+		for {
+			extractedLogs := extractLogs(srcReq.ld, capacityLeft)
+			if extractedLogs.LogRecordCount() == 0 {
+				break
+			}
+			capacityLeft -= extractedLogs.LogRecordCount()
+			if destReq == nil {
+				destReq = &logsRequest{ld: extractedLogs, pusher: srcReq.pusher}
+			} else {
+				extractedLogs.ResourceLogs().MoveAndAppendTo(destReq.ld.ResourceLogs())
+			}
+			if capacityLeft == 0 {
```

Comment:
Do we want to do `capacityLeft <= 0` check? If for whatever reason a regression is introduced in logs extractions prior, at least the data request isn't lost.

Reply:
Good point

Reply:
updated